runtime/drivers/blob/blobdownloader.go

```diff
@@ -426,12 +372,21 @@ func (it *blobIterator) plan() ([]*objectWithPlan, error) {
 		}
 	}
 
-	it.logger.Info("planner completed", zap.String("glob", it.opts.GlobPattern), zap.Int64("listed_objects", fetched), zap.Int("matched", matchCount), zap.Int64("bytes_matched", size), observability.ZapCtx(it.ctx))
-
 	items := planner.items()
 	if len(items) == 0 {
 		return nil, fmt.Errorf("no files found for glob pattern %q", it.opts.GlobPattern)
 	}
+
+	if size < it.opts.BatchSizeBytes { // need to ingest complete batch in one go
+		it.batchCount = _concurrentBlobDownloadLimit
+	} else {
+		// may be just keep it some factor of _concurrentBlobDownloadLimit ?
+		it.batchCount = int(it.opts.BatchSizeBytes/(size/int64(matchCount))) + 1
+	}
```

Comment:
Not fully sure what this is trying to accomplish – can we avoid the `batchCount` entirely? It would be easier to reason about if we just rely on batch size in bytes and the concurrent download limit.